### PR TITLE
Improved Documentation for basic.py to make it more consistent

### LINKED ISSUE #16627: TypeError on Or.subs()
Hi,

I'm trying to substitute variables in a Boolean expression but it raises a `TypeError`. I have this error both in 1.3 and 1.4 version of Sympy.

Thanks for your help!
Franck

```python
In [1]: import sympy                                                            

In [2]: e = sympy.Symbol("spam") | sympy.Symbol("foo")                          

In [3]: e.subs("foo", "bar", simultaneous=True)                                 
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
~/.local/lib/python3.6/site-packages/sympy/core/cache.py in wrapper(*args, **kwargs)
     93                 try:
---> 94                     retval = cfunc(*args, **kwargs)
     95                 except TypeError:

~/.local/lib/python3.6/site-packages/sympy/core/basic.py in _subs(self, old, new, **hints)
   1110         if rv is None:
-> 1111             rv = fallback(self, old, new)
   1112         return rv

~/.local/lib/python3.6/site-packages/sympy/core/basic.py in fallback(self, old, new)
   1087             if hit:
-> 1088                 rv = self.func(*args)
   1089                 hack2 = hints.get('hack2', False)

~/.local/lib/python3.6/site-packages/sympy/core/operations.py in __new__(cls, *args, **options)
    412             # without having to sympify all values
--> 413             _args = frozenset(cls._new_args_filter(args))
    414         except ShortCircuit:

~/.local/lib/python3.6/site-packages/sympy/logic/boolalg.py in _new_args_filter(cls, args)
    774         rel = []
--> 775         args = BooleanFunction.binary_check_and_simplify(*args)
    776         for x in args:

~/.local/lib/python3.6/site-packages/sympy/logic/boolalg.py in binary_check_and_simplify(self, *args)
    449         from sympy.core.relational import Relational, Eq, Ne
--> 450         args = [as_Boolean(i) for i in args]
    451         bin = set().union(*[i.binary_symbols for i in args])

~/.local/lib/python3.6/site-packages/sympy/logic/boolalg.py in <listcomp>(.0)
    449         from sympy.core.relational import Relational, Eq, Ne
--> 450         args = [as_Boolean(i) for i in args]
    451         bin = set().union(*[i.binary_symbols for i in args])

~/.local/lib/python3.6/site-packages/sympy/logic/boolalg.py in as_Boolean(e)
     52         return e
---> 53     raise TypeError('expecting bool or Boolean, not `%s`.' % e)
     54 

TypeError: expecting bool or Boolean, not `_Dummy_19*_Dummy_20`.

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
<ipython-input-3-a3f29c4f1997> in <module>
----> 1 e.subs("foo", "bar", simultaneous=True)

~/.local/lib/python3.6/site-packages/sympy/core/basic.py in subs(self, *args, **kwargs)
    986                 # in things like Derivative(f(x, y), x) in which x
    987                 # is both free and bound
--> 988                 rv = rv._subs(old, d*m, **kwargs)
    989                 if not isinstance(rv, Basic):
    990                     break

~/.local/lib/python3.6/site-packages/sympy/core/cache.py in wrapper(*args, **kwargs)
     94                     retval = cfunc(*args, **kwargs)
     95                 except TypeError:
---> 96                     retval = func(*args, **kwargs)
     97                 return retval
     98 

~/.local/lib/python3.6/site-packages/sympy/core/basic.py in _subs(self, old, new, **hints)
   1109         rv = self._eval_subs(old, new)
   1110         if rv is None:
-> 1111             rv = fallback(self, old, new)
   1112         return rv
   1113 

~/.local/lib/python3.6/site-packages/sympy/core/basic.py in fallback(self, old, new)
   1086                     args[i] = arg
   1087             if hit:
-> 1088                 rv = self.func(*args)
   1089                 hack2 = hints.get('hack2', False)
   1090                 if hack2 and self.is_Mul and not rv.is_Mul:  # 2-arg hack

~/.local/lib/python3.6/site-packages/sympy/core/operations.py in __new__(cls, *args, **options)
    411             # is done so short-circuiting can be done
    412             # without having to sympify all values
--> 413             _args = frozenset(cls._new_args_filter(args))
    414         except ShortCircuit:
    415             return sympify(cls.zero)

~/.local/lib/python3.6/site-packages/sympy/logic/boolalg.py in _new_args_filter(cls, args)
    773         newargs = []
    774         rel = []
--> 775         args = BooleanFunction.binary_check_and_simplify(*args)
    776         for x in args:
    777             if x.is_Relational:

~/.local/lib/python3.6/site-packages/sympy/logic/boolalg.py in binary_check_and_simplify(self, *args)
    448     def binary_check_and_simplify(self, *args):
    449         from sympy.core.relational import Relational, Eq, Ne
--> 450         args = [as_Boolean(i) for i in args]
    451         bin = set().union(*[i.binary_symbols for i in args])
    452         rel = set().union(*[i.atoms(Relational) for i in args])

~/.local/lib/python3.6/site-packages/sympy/logic/boolalg.py in <listcomp>(.0)
    448     def binary_check_and_simplify(self, *args):
    449         from sympy.core.relational import Relational, Eq, Ne
--> 450         args = [as_Boolean(i) for i in args]
    451         bin = set().union(*[i.binary_symbols for i in args])
    452         rel = set().union(*[i.atoms(Relational) for i in args])

~/.local/lib/python3.6/site-packages/sympy/logic/boolalg.py in as_Boolean(e)
     51     if isinstance(e, Boolean):
     52         return e
---> 53     raise TypeError('expecting bool or Boolean, not `%s`.' % e)
     54 
     55 

TypeError: expecting bool or Boolean, not `_Dummy_19*_Dummy_20`.
```

### PR DESCRIPTION
**Pull request by Sc0rpi0n101**
#### Brief description of what is fixed or changed
* Improved Documentation to make it consistent throughout basic.py

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by Sc0rpi0n101** [comment]
Test1:
```python
>>> classify_ode(f(x).diff(x)**2, f(x))
('nth_algebraic', 'lie_group', 'nth_algebraic_Integral')
>>> classify_ode(f(x).diff(x)**2, f(x)) == ('nth_algebraic', 'lie_group', 'nth_algebraic_Integral')
True
>>> assert classify_ode(f(x).diff(x)**2, f(x)) == ('nth_algebraic', 'lie_group', 'nth_algebraic_Integral')
```

Test 2:

```python
>>> checkodesol(eq1a, sol1a, order=2, solve_for_func=False)
(True, 0)
>>> assert checkodesol(eq1a, sol1a, order=2, solve_for_func=False)[0]
```
test 3:

```python
>>> checkodesol(eqn, sol, order=2, solve_for_func=False)
(True, 0)
>>> assert checkodesol(eqn, sol, order=2, solve_for_func=False)[0]
```


```python
>>> test("sympy/solvers/tests/test_ode.py")
====================================== test process starts =======================================
executable:         /home/sc0rpi0n101/anaconda3/envs/sympy-dev-py35/bin/python  (3.7.1-final-0) [CPython]
architecture:       64-bit
cache:              yes
ground types:       python 
numpy:              1.16.2
random seed:        42419623
hash randomization: on (PYTHONHASHSEED=3783320180)

sympy/solvers/tests/test_ode.py[116] .........w..f..www.........ww.w.....w.f.....w..wff.w.fww.....
........w.........w.....f.wf..w.........w.....f.f.fff..                                       [OK]

_________________________________________ slowest tests __________________________________________
test_separable3 - Took 10.111 seconds
test_nth_order_linear_euler_eq_nonhomogeneous_variation_of_parameters - Took 10.188 seconds
test_order_reducible - Took 21.018 seconds
========= tests finished: 86 passed, 18 skipped, 12 expected to fail, in 208.08 seconds ==========
True
```

I don't know why or how they're failing in Travis.
@sylee957  @asmeurer  any idea?

**Comment by oscarbenjamin** [comment]
I suggest to split this into two PRs. In this PR just do the changes to docstrings. The removal of m in basic.py is the only effective change and has probably caused the test failures. Put that change in a separate PR (and the tests if they are connected to that change).

**Comment by Sc0rpi0n101** [comment]
@sylee957  please change the label to documentation as I've created another pull request for changes in subs as suggested by Oscar and this one only has documentation changes.

File: sympy/core/tests/test_subs.py [hunk 1]
```diff
@@ -640,7 +640,7 @@ def test_simultaneous_subs():
     assert (x/y).subs(reps) != (y/x).subs(reps)
     assert (x/y).subs(reps, simultaneous=True) == \
         (y/x).subs(reps, simultaneous=True)
-    assert Derivative(x, y, z).subs(reps, simultaneous=True) == \
```
Comment (asmeurer) [inline sympy/core/tests/test_subs.py]: This change seems unrelated to this PR.

**Comment by asmeurer** [comment]
I'm not sure if it necessarily helps readability to have empty lines between all the code examples.

File: sympy/core/tests/test_subs.py (hunk 1 above)
Comment (Sc0rpi0n101) [reply sympy/core/tests/test_subs.py]: Yeah sorry, I didn't remove it when I removed the other changes.
I will remove it as it is included in the other PR.

**Comment by Sc0rpi0n101** [comment]
It might not always, but I think sometimes it is better looking and more readable to have space after the outputs and separating the examples than having all the code and output cramped up.

File: sympy/core/tests/test_subs.py (hunk 1 above)
Comment (Sc0rpi0n101) [reply sympy/core/tests/test_subs.py]: Done

**Comment by oscarbenjamin** [comment]
Looking through the changes here I'm not sure how useful they are.

A lot of changes are reformatting that I don't necessary agree with. There are some new docstrings but they mostly don't convey much information...

**Comment by smichr** [comment]
I agree with the other assessments of the changes. These changes may also interfere with other PRs without adding much value while potentially increasing the work that other will have to do. For both reasons, please accept a friendly closure of this issue.

File: sympy/core/basic.py [hunk 2]
```diff
@@ -760,24 +807,25 @@ def _sorted_args(self):
...
-           >>> print((x**2 + x*y).as_poly())
-           Poly(x**2 + x*y, x, y, domain='ZZ')
+        >>> from sympy import sin
+        >>> from sympy.abc import x, y
+        >>> print((x**2 + x*y).as_poly())
+        Poly(x**2 + x*y, x, y, domain='ZZ')
 
-           >>> print((x**2 + x*y).as_poly(x, y))
-           Poly(x**2 + x*y, x, y, domain='ZZ')
+        >>> print((x**2 + x*y).as_poly(x, y))
```
Comment (sylee957) [inline sympy/core/basic.py]: I think that the changes like these can still be useful.
Can you open up a new PR containing only the changes about the bad indentation like these?

Comment (Sc0rpi0n101) [reply sympy/core/basic.py]: Yes I will
